cql/DashboardLibrary.cql

```diff
@@ -362,56 +453,56 @@ define PertinentMedicationStatements:
 define PertinentMedicationRequestsSummary:
   (PertinentMedicationRequests) M
   return {
-    Type: 'Request',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Common.DateTimeText(M.authoredOn),
-    End:   null as String, // MedicationRequest end date not supported in R4
-    Status: M.status.value
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Common.DateTimeText(M.authoredOn),
+    reference: 'MedicationRequest/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
 
 define PertinentMedicationDispensesSummary:
   (PertinentMedicationDispenses) M
   return {
-    Type: 'Dispense',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Coalesce(
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Coalesce(
       Common.DateTimeText(M.whenHandedOver),
       Common.DateTimeText(M.whenPrepared)
     ),
-    End:   null as String, // MedicationDispense end date not supported in R4
-    Status: M.status.value
+    reference: 'MedicationDispense/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
 
 define PertinentMedicationAdministrationsSummary:
   (PertinentMedicationAdministrations) M
   return {
-    Type: 'Administration',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Common.DateTimeText((M.effective as FHIR.Period)."start"),
-    End:   Common.DateTimeText((M.effective as FHIR.Period)."end"),
-    Status: M.status.value
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Common.DateTimeText((M.effective as FHIR.Period)."start") +
+      ' to ' +
+      Common.DateTimeText((M.effective as FHIR.Period)."end"),
+    reference: 'MedicationAdministration/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
 
 define PertinentMedicationStatementsSummary:
   (PertinentMedicationStatements) M
   return {
-    Type:  'Statement',
-    Name:  Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Common.DateTimeText((M.effective as FHIR.Period)."start"),
-    End:   Common.DateTimeText((M.effective as FHIR.Period)."end"),
-    Status: M.status.value
+    name:  Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Common.DateTimeText((M.effective as FHIR.Period)."start") +
+      ' to ' +
```

Comment:
@dwwinters Should we include a user-friendly way to handle the date text if only a start time exists (ex. for an active medication). What about for a date represented as a DateTime versus as a Period?

(NOTE: This applies to MedicationAdministration as well.)